cmd/agent/app/reporter/grpc/reporter.go

```diff
@@ -32,6 +32,7 @@ import (
 
 // Reporter reports data to collector over gRPC.
 type Reporter struct {
+	conn      *grpc.ClientConn
```

Comment:
I am not sure about this. If Reporter fully owns the grpc connection, then there is no need to pass it externally, it can create the connection itself. Since we're passing it externally (perhaps because it is shared for another functionality, like running both span reporting and sampling strategies via the same connection), then reporter should not be in charge of closing the connection, it should be done by whoever created the connection.

Reply:
This was also something I wasn't happy. The other choice would be to store the proxy as a property of the agent. On a second thought, it does make sense to have it there, so that the agent can close this connection once it's stopped.